shared/src/main/java/com/couchbase/lite/AbstractReplicator.java

```diff
@@ -557,6 +565,17 @@ private void _start() {
         boolean pull = isPull(config.getReplicatorType());
         boolean continuous = config.isContinuous();
 
+        C4ReplicationFilter filter = new C4ReplicationFilter() {
```

Comment:
I think we should check if there is at least a pushFilter or pullFilter in the config before creating the filter object.